src/main/business/UserManager/UserController.js

```diff
@@ -8,22 +8,24 @@ const { USER_PROCESS_ERROR_MSGS: ERROR_MSGS } = require("../../ErrorMessages");
 
 class UserController {
     #userRepo;
-    #sessionManager
 
-    constructor(userRepo, sessionManager){
-        this.#sessionManager = sessionManager;
+    constructor(userRepo){
         this.#userRepo = userRepo;
     }
 
+
+    #verifyType(type, desiredType){
+        if(type !== desiredType) throw new EMSError(ERROR_MSGS.INCORRECT_TYPE(type,desiredType), ERROR_CODES.INCORRECT_TYPE);
+    }
+
     /**
      * Registers a new user to the system
-     * @param session Admin Session
      * @param userDetails All details needed to register the new User.
      * @return Business instance of the newly registered user
      * @note The newly registered user is created with a default password.
      */
-    async register(session, userDetails) {
-        await this.verifyType(session, USER_TYPES.ADMIN);
+    async register(userDetails) {
+        this.#verifyType(userDetails.callingUser.type, USER_TYPES.ADMIN)
         this.#verifyUserDetails(userDetails);
         userDetails.password = DEFAULT_PASSWORD;
         const dalUser = await this.#userRepo.addUser(userDetails);
```

Comment:
Now the details sent to the DAL layer contains extra information. 
Please run a test to make sure this doesn't affect behavior.

Reply:
But does it still work?

Reply:
If you have a good reason (u see i change an argument name and didnt pass it, or wrap something in an object), than I will check it.

I dont even know that it worked before....

Reply:
> If it worked before it will work now.
We both don't know that because we both didn't test it. 

> I dont even know that it worked before....
It did work before and there are tests to prove that. 
Please run test that pushes extra fields to the userRepo and see that the user is added